### DISCUSSION TITLE
Suggestion for C# exception throw syntactical sugar

### DISCUSSION THREAD

**Discussion by mschwa** [dotnet/csharplang#326] [General]
This thought just struck me. Not entirely sure where to suggest it so I'm doing it here. If this is the wrong place I'm totally sorry, just let me know and I will never do it again ;-) 

Anyway. It would be super nice not to have to type "new" every time a throw an exception. Is there a case where you are not creating an object reference for an exception? 

**Version Used**: 

C# <= 7

**Current Behavior**:

1.  throw an exception

`throw new InvalidOperationException("Email must be present.");`

**Suggested Behavior**:

`throw InvalidOperationException("Email must be present.");`

**Comment by HaloFour** [comment]
`throw InvalidOperationException("Email must be present.");`

That is already legal syntax where `InvalidOperationException` is a method in scope.  This proposal would create an ambiguity there for seemingly little reason other than to avoid a couple keystrokes.

**Comment by DavidArno** [comment]
@HaloFour,

We have seen a number of proposals that each reduces the amount of "noise" in C# code by a few characters. Each taken on their own, adds little. Taken together though, they could add up to a significant effect.

However, I don't think this proposal passes the "readablility over writeability" test. The optional nature of `new` in F# is one of the features I don't like about that language, for example. It doesn't reduce noise: it makes the code easier to write (ergonomics as @CyrusNajmabadi calls it), but makes it more difficult to read. So I don't like this proposal either.
